crates/api_crud/src/site/read.rs

```diff
@@ -21,46 +22,67 @@ use lemmy_utils::{
+  let mut site_response = CACHE
+    .try_get_with::<_, LemmyError>((), async {
+      let site_view = SiteView::read_local(&mut context.pool()).await?;
+      let admins = PersonView::admins(&mut context.pool()).await?;
+      let all_languages = Language::read_all(&mut context.pool()).await?;
+      let discussion_languages = SiteLanguage::read_local_raw(&mut context.pool()).await?;
+      let taglines = Tagline::get_all(&mut context.pool(), site_view.local_site.id).await?;
+      let custom_emojis =
+        CustomEmojiView::get_all(&mut context.pool(), site_view.local_site.id).await?;
+      Ok(GetSiteResponse {
+        site_view,
+        admins,
+        version: version::VERSION.to_string(),
+        my_user: None,
+        all_languages,
+        discussion_languages,
+        taglines,
+        custom_emojis,
+      })
+    })
+    .await
+    .map_err(|e| anyhow::anyhow!("Failed to construct site response: {e}"))?;
 
-  // Build the local user
-  let my_user = if let Some(local_user_view) = local_user_view {
+  // Build the local user with parallel queries and add it to site response
+  site_response.my_user = if let Some(local_user_view) = local_user_view {
     let person_id = local_user_view.person.id;
     let local_user_id = local_user_view.local_user.id;
 
-    let follows = CommunityFollowerView::for_person(&mut context.pool(), person_id)
-      .await
-      .with_lemmy_type(LemmyErrorType::SystemErrLogin)?;
-
-    let person_id = local_user_view.person.id;
-    let community_blocks = CommunityBlockView::for_person(&mut context.pool(), person_id)
-      .await
-      .with_lemmy_type(LemmyErrorType::SystemErrLogin)?;
-
-    let instance_blocks = InstanceBlockView::for_person(&mut context.pool(), person_id)
-      .await
-      .with_lemmy_type(LemmyErrorType::SystemErrLogin)?;
-
-    let person_id = local_user_view.person.id;
-    let person_blocks = PersonBlockView::for_person(&mut context.pool(), person_id)
-      .await
-      .with_lemmy_type(LemmyErrorType::SystemErrLogin)?;
-
-    let moderates = CommunityModeratorView::for_person(&mut context.pool(), person_id)
-      .await
-      .with_lemmy_type(LemmyErrorType::SystemErrLogin)?;
-
-    let discussion_languages = LocalUserLanguage::read(&mut context.pool(), local_user_id)
-      .await
-      .with_lemmy_type(LemmyErrorType::SystemErrLogin)?;
+    let (
+      follows,
+      community_blocks,
+      instance_blocks,
+      person_blocks,
+      moderates,
+      discussion_languages,
+    ) = try_join!(
+      CommunityFollowerView::for_person(context.inner_pool(), person_id),
+      CommunityBlockView::for_person(context.inner_pool(), person_id),
+      InstanceBlockView::for_person(context.inner_pool(), person_id),
+      PersonBlockView::for_person(context.inner_pool(), person_id),
+      CommunityModeratorView::for_person(context.inner_pool(), person_id),
+      LocalUserLanguage::read(context.inner_pool(), local_user_id)
+    )
```

Comment:
Instead of changing the functions to use ActualDbPool, you can replace this with:

```
lemmy_db_schema::try_join_with_pool!(pool => (
       |pool| CommunityFollowerView::for_person(pool, person_id),
       |pool| CommunityBlockView::for_person(pool, person_id),
       |pool| InstanceBlockView::for_person(pool, person_id),
       |pool| PersonBlockView::for_person(pool, person_id),
       |pool| CommunityModeratorView::for_person(pool, person_id),
       |pool| LocalUserLanguage::read(pool, local_user_id)
     ))
```

Pool variable must be defined first because the macro only accepts an identifier:

```
let pool = context.pool();
```

The queries for building site_response can also run in parallel.

Reply:
Nice I didnt see that macro, works perfectly.